File: src/problems/H_Index.java
package problems;

import java.util.Arrays;

public class H_Index {

    public int solution(int[] citations) {
        Arrays.sort(citations);
        int size = citations.length;

        int result = 0;
        for (int i = size - 1; i >= 0; i--) {
            if (citations[i] >= size - i) {
                result++;
            } else break;
        }
        return result;
    }
}

File: src/problems/HIndex.kt
package problems

class HIndex {
    fun hIndex(citations: IntArray): Int {
        citations.sortDescending()
        var result = 0
        for (i in 0..citations.size - 1) {
            if (i + 1 <= citations[i]) {
                result = i + 1
            }
        }
        return result
    }
}